infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -135,107 +142,103 @@ public void checkpointState() {
         }
     }
 
-    @Override
-    public synchronized void handleMessage(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        switch (msg.getMsgType()) {
-            case TOKEN_REQ: {
-                TokenRequest req = ((CorfuPayloadMsg<TokenRequest>) msg).getPayload();
-                if (req.getNumTokens() == 0) {
-                    long max = 0L;
-                    boolean hit = false;
-                    ImmutableMap.Builder<UUID, Long> streamsLastIssued = ImmutableMap.builder();
+    /**
+     * Service an incoming token request.
+     */
+    @ServerHandler(type=CorfuMsgType.TOKEN_REQ)
+    public synchronized void tokenRequest(CorfuPayloadMsg<TokenRequest> msg,
+                                          ChannelHandlerContext ctx, IServerRouter r) {
+        TokenRequest req = msg.getPayload();
+        if (req.getNumTokens() == 0) {
+            long max = 0L;
+            boolean hit = false;
+            ImmutableMap.Builder<UUID, Long> streamsLastIssued = ImmutableMap.builder();
+            for (UUID id : req.getStreams()) {
```

Comment:
Oops, I just deleted my comment because after posting I (mistakenly?) noticed that this pull request had been submitted... I should have waited for this PR to be submitted before sending mine, right? If so, you can delete my PR and I will wait for this PR to be merged.